core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -690,8 +706,9 @@ object KafkaConfig {
       .define(LeaderImbalancePerBrokerPercentageProp, INT, Defaults.LeaderImbalancePerBrokerPercentage, HIGH, LeaderImbalancePerBrokerPercentageDoc)
       .define(LeaderImbalanceCheckIntervalSecondsProp, LONG, Defaults.LeaderImbalanceCheckIntervalSeconds, HIGH, LeaderImbalanceCheckIntervalSecondsDoc)
       .define(UncleanLeaderElectionEnableProp, BOOLEAN, Defaults.UncleanLeaderElectionEnable, HIGH, UncleanLeaderElectionEnableDoc)
-      .define(InterBrokerSecurityProtocolProp, STRING, Defaults.InterBrokerSecurityProtocol, MEDIUM, InterBrokerSecurityProtocolDoc)
+      .define(InterBrokerSecurityProtocolProp, STRING, null, MEDIUM, InterBrokerSecurityProtocolDoc)
```

Comment:
Yeah, it's not ideal, but I am not sure how to implement the desired behaviour otherwise: it should default to PLAINTEXT if the user hasn't set it, but it should give an error if the user has set `inter.broker.listener.name` and `security.inter.broker.protocol`. If we leave the `PLAINTEXT` default in the config definition, it seems like we cannot distinguish the case where the user set it versus the default. Hmm, maybe we can use `originals()` for this, I'll try it.